falcon_kit/stats_preassembly.py

```diff
@@ -115,7 +116,25 @@ def read_lens_from_db(db_fn):
     """
     return list(sorted(get_db_readlengths(db_fn)))
 
-def stats_dict(stats_raw_reads, stats_seed_reads, stats_corrected_reads, genome_length, length_cutoff):
+def metric_fragmentation(preads_dir):
+    # https://jira.pacificbiosciences.com/browse/SAT-105
+    bash = r"""
+sed -nr 's;>prolog/([0-9]*)[0-9]/.*;\1;p' %s/*.fasta | sort | uniq -c | awk '{print $1}' | sort | uniq -c
```

Comment:
That's fine with me. You're worried about the runtime of the first sort? How slow was it? Did you measure it in isolation, or is it possible that sed on the gigantic files was actually the slow part?

sed is much faster than python at parsing. If speed is the issue, then perl is a better choice than python. Any objection?